File: practice22/4.cpp

//https://www.nowcoder.com/pat/6/problem/4050
#include<iostream>
#include<string>
#include<algorithm>
using namespace std;

void science_count1(string& str)
{
	str.erase(1, 1);
	int x = 0, sum = 0, n, m, sign = 0;
	while (1)
	{
		if (str[x] == '-' || str[x] == '+')
		{
			break;
		}
		x++;
	}
	if (str[x] == '+')
	{
		sign = 1;
	}
	str.erase(x - 1, 2);
	n = str.size() - x + 1;
	x = x - 2;
	for (int i = x + 1; i < str.size(); i++)
	{
		sum = sum + (str[i] - '0')*pow(10, --n);
	}
	str.erase(x + 1, str.size() - x - 1);
	if (sign == 1)
	{
		if (sum < str.size() - 1)
		{
			str.insert(str.begin() + str.size() - sum + 1, '.');
		}
		else
		{
			str.insert(str.begin() + str.size(), sum - str.size() + 1, '0');
		}
		cout << str << endl;
	}
	else
	{
		cout << "0.";
		for (int i = 0; i < sum - 1; i++)
		{
			cout << '0';
		}
		cout << str << endl;
	}
}

int main()
{
	string str;
	while (cin >> str)
	{
		if (str[0] == '-')
		{
			cout << '-';
		}
		str.erase(0, 1);
		science_count1(str);
	}
	system("pause");
	return 0;
}

File: practice29/4.java



//https://www.nowcoder.com/pat/2/problem/359
import java.util.Scanner;
public class Main{
    public static void main(String []args){
        Scanner scanner=new Scanner(System.in);
        String []str={"星期天","星期一","星期二","星期三","星期四","星期五","星期六"};
        int []p1={31,29,31,30,31,30,31,31,30,31,30,31};
        int []p2={31,28,31,30,31,30,31,31,30,31,30,31};
        while(scanner.hasNext()) {
            int n = scanner.nextInt();
            int[] pt = new int[12];
            if (n % 4 == 0 && n % 100 != 0 || n % 400 == 0) {
                pt = p1;
            } else {
                pt = p2;
            }
            int sum = 13;
            int result, x;
            int sign=0;
            for (int i = 0; i < 12; i++) {
                result = n - 1 + (n - 1) / 4 - (n - 1) / 100 + (n - 1) / 400 + sum;
                x = result % 7;
                if (str[x] == "星期五") {
                    if(sign==1){
                        System.out.printf("%c%d",' ',i+1);
                    }
                    else{
                        System.out.printf("%d",i+1);
                    }
                    sign=1;
                }
                sum = sum + pt[i];
            }
            System.out.println();
        }
    }
}
